edgepi-leds.html

```diff
@@ -1,167 +1,199 @@
-        },
-        inputs:1,
-        outputs:1,
-        icon: "light.svg",
-        label: function() {
-            return this.name||"led";
-        },
-        oneditprepare: function(){
-            // Query html elements
-            
-            const configTransportSelect = document.getElementById("node-input-transport");
-            const tcpField = document.querySelector(".form-row.tcp")
-            const configSelect = document.getElementById("node-input-config");
-            const methodField = document.querySelector(".form-row.method");
-            const ledPinField = document.querySelector(".form-row.ledPin");
+  RED.nodes.registerType("led", {
+    category: "EdgePi",
+    color: "#f391aa",
+    defaults: {
+      name: { value: "" },
+      transport: { value: "Local" },
+      ledState: { value: "turnOn" },
+      ledPin: { value: "LED1" },
+      tcpAddress: { value: "" },
+      tcpPort: { value: "" },
+    },
+    inputs: 1,
+    outputs: 1,
+    icon: "light.svg",
+    label: function () {
+      return this.name || "led";
+    },
+    oneditprepare: function () {
+      const configTransportSelect = document.getElementById(
+        "node-input-transport"
+      );
+      const tcpField = document.querySelector(".form-row.tcp");
+      const ledState = document.querySelector(".form-row.ledState");
+      const ledPinField = document.querySelector(".form-row.ledPin");
+      const ledOn = document.getElementById("node-input-turnOn");
+      const ledOff = document.getElementById("node-input-turnOff");
 
-            function updateEditor(){
-                tcpField.style.display = (configTransportSelect.value === "Network") ? "flex" : "none";
-                methodField.style.display = (configSelect.value === "Editor") ? "flex" : "none";
-                ledPinField.style.display = (configSelect.value === "Editor") ? "flex" : "none";
-            }
-            // Update and add event listener to update on user input
-            updateEditor()
-            configTransportSelect.addEventListener("change", updateEditor);
-            configSelect.addEventListener("change", updateEditor);
-        }
-    });
+      if (this.ledState == "turnOn") {
+        ledOn.checked = true;
+      } else {
+        ledOff.checked = true;
+      }
+
+      function updateEditor() {
+        tcpField.style.display =
+          configTransportSelect.value === "Network" ? "flex" : "none";
+      }
+      updateEditor();
+      configTransportSelect.addEventListener("change", updateEditor);
+      ledOn.addEventListener("change", updateEditor);
+      ledOff.addEventListener("change", updateEditor);
+    },
+    oneditsave: function () {
+      this.ledState = document.getElementById("node-input-turnOn").checked
+        ? "turnOn"
+        : "turnOff";
+    },
+  });
 </script>
 
 <script type="text/html" data-template-name="led">
-    <style>
-        *{
-            box-sizing: border-box !important;
-        }
+  <style>
```

Comment:
Why is <style> tag in the <script> tag? <style> tag should be in the <head> tag of the HTML file